spec/requests/api/v1/moves_controller_create_spec.rb

```diff
@@ -201,5 +201,43 @@
 
       it_behaves_like 'an endpoint that responds with error 422'
     end
+
+    context 'with a duplicate move', :skip_before do
+      let(:profile) { create(:profile) }
+      let(:person) { profile.person }
+      let(:move_attributes) do
+        attributes_for(:move).merge(date: old_move.date,
+                                    person: person,
+                                    from_location: from_location,
+                                    to_location: to_location)
+      end
+
+      before do
+        post '/api/v1/moves', params: { data: data, access_token: token.token }, as: :json
+      end
+
+      context 'when duplicate is cancelled' do
+        let!(:old_move) { create(:move, :cancelled, person: person, from_location: from_location, to_location: to_location) }
+
+        it_behaves_like 'an endpoint that responds with success 201'
+      end
+
+      context 'when duplicate is active' do
+        let!(:old_move) { create(:move, person: person, from_location: from_location, to_location: to_location) }
+        let(:errors_422) do
+          [
+            {
+              title: 'Unprocessable entity',
+              detail: 'Date has already been taken',
```

Comment:
Is Date our only duplicate error?

Reply:
Yes - you're right it is a little fragile from that point of view. Can't think off-hand how to deal with that.

Reply:
@teneightfive Amended implementation to be more generic - this can now support any kind of uniqueness check, the only difference being in the `meta` key below which now uses `existing_id` rather than `move_id` so that the same approach can be extended to other models in future.

For now, only moves will return this error and they only have a uniqueness check on the date attribute.